src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -949,6 +949,11 @@ void* ClassLoader::dll_lookup(void* lib, const char* name, const char* path) {
 
 void ClassLoader::load_java_library() {
   assert(CanonicalizeEntry == nullptr, "should not load java library twice");
+  if (is_vm_statically_linked()) {
+    CanonicalizeEntry = CAST_TO_FN_PTR(canonicalize_fn_t, os::lookup_function("JDK_Canonicalize"));
```

Comment:
Can you add an assert to make sure `CanonicalizeEntry` is not NULL?

Reply:
Also please remember to use nullptr and not NULL! @kimbarrett would appreciate it :)